### PR TITLE
add:added wishlist page

### PR DESCRIPTION
**Pull request by JuvitaSaini09**
### Files to be checked:

- wishlist.js
- wishlistContext.js
- wishlistPage.css
- wishlistPage.js

Snapshot
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import "./App.css";
 import { Home } from "./pages/home/home";
 import { ProductPage } from "./pages/productPage/productPage";
 import { CartPage } from "./pages/cartPage/cartPage";
+import { WishlistPage } from "./pages/wishlistPage/wishlistPage";
```
Comment (tushartiwari7) [inline src/App.js]: Suggestion: you may export all files from `./index.js`

File: src/context/wishlistContext/wishlistContext.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import React, { createContext, useContext, useReducer } from 'react'
+
+const wishlistContext=createContext(null);
+
+const removeFromWishlist=(clickedItem,wishlistItems)=>{
+    
+     return wishlistItems.filter(item=>item._id!==clickedItem._id)
+  
+}
+
```
Comment (tushartiwari7) [inline src/context/wishlistContext/wishlistContext.js]: use prettier

File: src/components/wishlist/wishlist.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+function Wishlist() {
+ const {dispatch}=useCart();
+  const  {stateWishlist,dispatchWishList}=useWishlist();
+  return (
+    <>
+  <div>
+  <h1 className="heading-my-wishlist">My Wishlist ❤</h1>
+  <section className="wishlist-card-container">
+    <>
```
Comment (gauravsinhaweb) [inline src/components/wishlist/wishlist.js:12]: react fragment is not required here!

File: src/context/wishlistContext/wishlistContext.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+const addToWishlist=(clickedItem,cart)=>{
+    const isItemInWishlist = cart.find((cartItem) => clickedItem._id === cartItem._id);
+     if(isItemInWishlist){
+      return cart.filter(item=>item._id!==clickedItem._id)
+     }
+     else{
```
Comment (tushartiwari7) [inline src/context/wishlistContext/wishlistContext.js]: noneed to put else here.

File: src/components/cart/cartItems.js [hunk 5]
```diff
@@ -1,11 +1,14 @@
 import React from 'react'
 import { useCart } from '../../context/cartContext/cartContext';
+import { useWishlist } from '../../context/wishlistContext/wishlistContext';
```
Comment (Ankur9669) [inline src/components/cart/cartItems.js:6]: Can remove unnessacry space here

File: src/pages/wishlistPage/wishlistPage.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import { Footer, NavbarLogout, Wishlist } from '../../components/allComponents'
+import "./wishlistPage.css"
+function WishlistPage() {
+  return (
+    <>
+    <NavbarLogout />
```
Comment (gauravsinhaweb) [inline src/pages/wishlistPage/wishlistPage.js:7]: NavbarLogout can be just Logout!

**Review by tushartiwari7 — APPROVED** [review]
Added Some suggestions, Please have a look.

**Review by gauravsinhaweb — APPROVED** [review]
UI looks good, Good to merge after changes!

File: src/components/wishlist/wishlist.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  {stateWishlist[0] === undefined ?<h1>Wishlist is Empty</h1>:""}
+  </>
+      { 
+        stateWishlist.map(item=>{
+          return(
+            <div className="wishlist-card">
```
Comment (Ankur9669) [inline src/components/wishlist/wishlist.js]: Use key here

File: src/context/wishlistContext/wishlistContext.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import React, { createContext, useContext, useReducer } from 'react'
+
+const wishlistContext=createContext(null);
+
+const removeFromWishlist=(clickedItem,wishlistItems)=>{
+    
+     return wishlistItems.filter(item=>item._id!==clickedItem._id)
+  
+}
```
Comment (Ankur9669) [inline src/context/wishlistContext/wishlistContext.js]: Unessacery spacing should be removed

**Review by Ankur9669 — APPROVED** [review]
The code looks good, but please use prettier code formatter so that the code becomes more readable

File: src/App.js (hunk 1 above)
Comment (JuvitaSaini09) [reply src/App.js]: thanks for suggestion,i have updated it

File: src/context/wishlistContext/wishlistContext.js (hunk 2 above)
Comment (JuvitaSaini09) [reply src/context/wishlistContext/wishlistContext.js]: updated

File: src/components/wishlist/wishlist.js (hunk 7 above)
Comment (JuvitaSaini09) [reply src/components/wishlist/wishlist.js]: updated, thnks for the suggestion

File: src/context/wishlistContext/wishlistContext.js (hunk 8 above)
Comment (JuvitaSaini09) [reply src/context/wishlistContext/wishlistContext.js]: ok

**Pull request merged**